goalie/mesh_seq.py

```diff
@@ -49,7 +48,7 @@ def __init__(self, time_partition, initial_meshes, **kwargs):
         :type parameters: :class:`~.AdaptParameters`
         """
         self.time_partition = time_partition
-        self.fields = time_partition.fields
+        self.fields = {field: None for field in time_partition.fields}
```

Comment:
Is the intention for this to hold non-prognostic fields as well? I'm conscious that with this change we will be using the term `fields` in place of `solutions`.

Also, if we do adopt `MeshSeq.fields` then `TimePartition.fields` should probably be renamed for clarity, e.g., `TimePartition.field_names`.

Reply:
That's a good point. The dictionary `mesh_seq.fields` does exactly what you said: holds the solutions at the current timestep. Should we rename it to something more descriptive of that?

Reply:
But it should also be distinct from the `solutions` in `function_data.py`

Reply:
With my analogy between `MeshSeq.function_spaces` and `MeshSeq.fields` that I described in https://github.com/mesh-adaptation/goalie/issues/179#issuecomment-2079326965, maybe we could rename `MeshSeq.fields` to `MeshSeq.functions`? I think that fits nicely